Add Normal-inverse Gaussian distribution

davidanthoff opened the pull request:
This needs tests, but I'm at a loss what to do with the new test framework. Is there some explanation how to hook into that?

lindahua commented:
I am working on a major overhaul of the tests. 

You can just write a separate test at this point. I will try to incorporate it after the work is done.

davidanthoff commented:
I added tests. The ones for the moment are a bit clumsy, I computed the "correct" value in R by sampling and then computing the moments based on that sample, which seems not ideal. But I couldn't find any other function in R that would compute things like the mean of the distribution, and I did want to use code that I hadn't written, just avoid errors that I might repeat.

I also rebased and squashed, so if this passes CI I think this could be merged.

johnmyleswhite commented:
Looked good to me. Thanks for this!
